pkg/codegen/schema/schema.go

```diff
@@ -215,6 +214,19 @@ func (t *ObjectType) String() string {
 
 func (*ObjectType) isType() {}
 
+type ResourceType struct {
```

Comment:
How does one actually use `ResourceType`?

In each SDK code generator, we'll need some way to determine that a given `schema.Type` is a `ResourceType` that should be emitted as a reference to a resource (i.e. the name of the resource).

For example, this is what it looks like for generating references to object types in Python:

https://github.com/pulumi/pulumi/blob/7c258c77c37732de220db15c9709135a65780f6e/pkg/codegen/python/gen.go#L1476-L1480

I assume we'll want to do something similar for `ResourceType`?

Do we need to add a `resources` map to the `types` struct?

https://github.com/pulumi/pulumi/blob/7c258c77c37732de220db15c9709135a65780f6e/pkg/codegen/schema/schema.go#L874-L883

Or something else?

Reply:
My understanding is that `/resources` is a `TokenType`, so we already have logic to handle that. I may be mistaken on that. @pgavlin ?

Reply:
> How does one actually use ResourceType?

I was wondering this myself, since we don't seem to create these anywhere.

> My understanding is that /resources is a TokenType, so we already have logic to handle that. I may be mistaken on that. @pgavlin ?

I would have expected us to create a `ResourceType` rather than a `TokenType`, in which case we would need a new map in that struct. I'd imagine that `ResourceType` would be a thin wrapper over a `Resource`, since the latter already includes its token, etc.

Reply:
> How does one actually use `ResourceType`?

Now that I'm looking at it again, I'm not sure if the new type is actually needed

Reply:
> I would have expected us to create a `ResourceType` rather than a `TokenType`, in which case we would need a new map in that struct.

Ok, I'll take another crack at this